nimbus/src/updating.rs

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! This module implements the primitive functions to implement
+//! safe updating from the server.
+
+use crate::error::Result;
+use crate::persistence::{Database, StoreId, Writer};
+use crate::Experiments;
+
+const KEY_PENDING_UPDATES: &str = "pending-experiment-updates";
+
+pub fn stash_pending_updates(db: &Database, experiments: Experiments) -> Result<()> {
+    let mut writer = db.write()?;
+    db.get_store(StoreId::Updates)
+        .put(&mut writer, KEY_PENDING_UPDATES, &experiments)?;
+    writer.commit()?;
+    Ok(())
+}
+
+pub fn pop_pending_updates(db: &Database, writer: &mut Writer) -> Result<Option<Experiments>> {
+    let store = db.get_store(StoreId::Updates);
+    let experiments = store.get::<Experiments>(writer, KEY_PENDING_UPDATES)?;
+
+    if let Some(_) = experiments {
+        store.clear(writer)?;
+    }
+
+    Ok(experiments)
+}
```

Comment:
Use an `Option` here to distinguish between "no pending updates", and "updates of an empty list", i.e. all experiments ended.

Reply:
Might be worth adding this note as a comment? For bonus points, maybe mention why the caller would care? A quick look shows this is a micro-optimization, so it's probably worth calling out that there's no logical different between `None` and `Some(vec![])`?